# DataSet

We will start from the dataset of the capitals of the world available on the simplemaps website and we will only consider Swiss cities. 
A future job could be to extend the selection to foreign cities to support more enterprising young people who do not exclude the possibility of moving outside the homeland.
The dataset under consideration is as follows:

In [4]:
import pandas as pd # library for data analsysis

import urllib.request

from io import BytesIO
from zipfile import ZipFile
from urllib.request import urlopen

print('libs imported!')

#######################################################################################################

url = 'https://simplemaps.com/static/data/world-cities/basic/simplemaps_worldcities_basicv1.6.zip'

hdr = {'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.11 (KHTML, like Gecko) Chrome/23.0.1271.64 Safari/537.11',
       'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
       'Accept-Charset': 'ISO-8859-1,utf-8;q=0.7,*;q=0.3',
       'Accept-Encoding': 'none',
       'Accept-Language': 'en-US,en;q=0.8',
       'Connection': 'keep-alive'}

extensions = ('.csv')

request=urllib.request.Request(url, None, headers=hdr) #The assembled request

zipfile = ZipFile(BytesIO(urlopen(request).read()))
files_csv = [zipfile.open(file, 'r') for file in zipfile.namelist() if file.endswith(extensions)]
df_capitals = pd.concat([pd.read_csv(file_csv) for file_csv in files_csv])
zipfile.close()

print('\ndf_capitals shape: {}\n'.format(df_capitals.shape))
df_capitals.sort_values(by='id').head(10)

libs imported!

df_capitals shape: (15493, 11)



city     city_ascii      lat      lng      country iso2 iso3  \
828        Kandahār       Kandahar  31.6100  65.6949  Afghanistan   AF  AFG   
4923          Qalāt          Qalat  32.1123  66.8868  Afghanistan   AF  AFG   
6506      Sar-e Pul      Sar-e Pul  36.2154  65.9325  Afghanistan   AF  AFG   
3813   Pul-e Khumrī   Pul-e Khumri  35.9511  68.7011  Afghanistan   AF  AFG   
5141  Maḩmūd-e Rāqī  Mahmud-e Raqi  35.0167  69.3333  Afghanistan   AF  AFG   
2694         Ghaznī         Ghazni  33.5633  68.4178  Afghanistan   AF  AFG   
6423    Pul-e ‘Alam    Pul-e `Alam  33.9953  69.0227  Afghanistan   AF  AFG   
1731         Kunduz         Kunduz  36.7280  68.8725  Afghanistan   AF  AFG   
1090          Herāt          Herat  34.3300  62.1700  Afghanistan   AF  AFG   
3945       Asadābād       Asadabad  34.8660  71.1500  Afghanistan   AF  AFG   

     admin_name capital  population          id  
828    Kandahār   admin    715542.0  1004003059  
4923      Zābul   admin     12191.0  1004016690  
6506  Sar-e Pul   admin         NaN  1004047427  
3813    Baghlān   admin     56369.0  1004123527  
5141     Kāpīsā   admin      7407.0  1004151943  
2694     Ghaznī   admin    141000.0  1004167490  
6423      Lōgar   admin         NaN  1004180853  
1731     Kunduz   admin    259809.0  1004227517  
1090      Herāt   admin    481009.0  1004237782  
3945      Kunaṟ   admin     48400.0  1004251962

There are 11 columns and 15,493 rows. Let's look at the columns:
   - city: city name
   - city_ascii: city name in ascii encoding
   - lat: city latitude
   - lng: city longitude
   - country: the name of city's country
   - iso2: iso code of the country (two characters)
   - iso3: iso code of the country (three characters)
   - admi_name: The name of the highest level administration region of the city
   - capital: Blank string if not a capital, otherwise: 
       - primary - country's capital 
       - admin - first-level admin capital 
       - minor - lower-level admin capital
   - population: an estimate of the city's urban population
   - id: A 10-digit unique id of the city